oonimkall/runner.go

```diff
@@ -114,13 +111,13 @@ func (r *runner) Run(ctx context.Context) {
 	// TODO(bassosimone): honour context
 	// TODO(bassosimone): intercept all options we ignore
 
+	logger := newChanLogger(r.emitter, r.settings.LogLevel, r.out)
```

Comment:
I have changed `newChanLogger`'s second argument to take the log level directly because

1. the default is WARNING when the log level is not recognised

2. otherwise it felt very weird that the we were passing to a constructor the settings before "validating" them in hasUnsupportedSettings (the name of the method actually is slightly different than validate settings, hence the quotes, and the previous code could probably have been fine anyway, but I really wanted to remove ambiguity.)